### ISSUE TITLE
Cypress.config() does not return the complete configuration

### ISSUE THREAD

**Issue by andreaazzara** [cypress-io/cypress#509]
* Operating System: macOS Sierra
* Cypress Version: 0.19.2
* Browser/Browser Version: Chrome 58

### Are you requesting a feature or reporting a bug?
Possible bug

### Current behavior:
My cypress.json is

```js
{
  "projectId": "myId",
  "watchForFileChanges": false,
  "fileServerFolder": "www",
  "viewportWidth": 320,
  "viewportHeight": 568,
  "numTestsKeptInMemory": 5,
  "trashAssetsBeforeHeadlessRuns": false,
  "videoRecording": false,
  "defaultCommandTimeout": 30000,
  "pageLoadTimeout": 30000,
  "fixturesFolder": "/Users/some-custom-folder"
}
```

but Cypress.config() returns:

```js
{
    "isHeadless": false,
    "numTestsKeptInMemory": 5,
    "waitForAnimations": true,
    "animationDistanceThreshold": 5,
    "defaultCommandTimeout": 30000,
    "pageLoadTimeout": 30000,
    "requestTimeout": 5000,
    "responseTimeout": 30000,
    "xhrUrl": "__cypress/xhrs/",
    "baseUrl": null,
    "viewportWidth": 320,
    "viewportHeight": 568,
    "execTimeout": 60000,
    "screenshotOnHeadlessFailure": true,
    "namespace": "__cypress"
}
```
where some keys are missing (fixturesFolder, videoRecording...)

**Comment by brian-mann** [comment]
You are right, `Cypress.config()` does not return the whole set of configuration. 

This is because we generally only provide you the mutable configuration that you can change at runtime (in the browser). 

Something like `fixturesFolder` is not present because its static and cannot be dynamically changed. Even if we were to provide you the values, changing them in the browser at runtime would have no effect.

When Cypress boots up, the internal server interacts with these values and is not built to expect them to change.

**Comment by andreaazzara** [comment]
I see, However accessing `fixturesFolder` variable (even if read-only) could be useful (for example to be used with `cy.readFile`. Maybe `Cypress.config()` could return all the env variables and trying to set a read-only one could throw an error.

**Comment by brian-mann** [comment]
Released in [`1.2.0`](https://docs.cypress.io/guides/references/changelog.html#1-2-0).
